receiver/otlpjsonfilereceiver/file.go

```diff
@@ -0,0 +1,147 @@
+)
+
+const (
+	typeStr   = "otlpjsonfile"
+	stability = component.StabilityLevelAlpha
+	transport = "file"
+)
+
+// NewFactory creates a factory for file receiver
+func NewFactory() component.ReceiverFactory {
+	return component.NewReceiverFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithMetricsReceiverAndStabilityLevel(createMetricsReceiver, stability),
+		component.WithLogsReceiverAndStabilityLevel(createLogsReceiver, stability),
+		component.WithTracesReceiverAndStabilityLevel(createTracesReceiver, stability))
+}
+
+type cfg struct {
+	config.ReceiverSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	fileconsumer.Config     `mapstructure:",squash"`
+}
+
+func createDefaultConfig() config.Receiver {
+	return &cfg{
+		Config:           *fileconsumer.NewConfig(),
+		ReceiverSettings: config.NewReceiverSettings(config.NewComponentID(typeStr)),
+	}
+}
+
+type receiver struct {
+	input *fileconsumer.Input
+	id    config.ComponentID
+}
+
+func (f *receiver) Start(ctx context.Context, host component.Host) error {
+	storageClient, err := adapter.GetStorageClient(ctx, f.id, component.KindReceiver, host)
+	if err != nil {
+		return err
+	}
+	return f.input.Start(storageClient)
+}
+
+func (f *receiver) Shutdown(ctx context.Context) error {
+	return f.input.Stop()
+}
+
+func createLogsReceiver(_ context.Context, settings component.ReceiverCreateSettings, configuration config.Receiver, logs consumer.Logs) (component.LogsReceiver, error) {
+	logsUnmarshaler := plog.NewJSONUnmarshaler()
+	obsrecv := obsreport.NewReceiver(obsreport.ReceiverSettings{
+		ReceiverID:             configuration.ID(),
+		Transport:              transport,
+		ReceiverCreateSettings: settings,
+	})
+	input, err := configuration.(*cfg).Config.Build(settings.Logger.Sugar(), func(ctx context.Context, attrs *fileconsumer.FileAttributes, token []byte) {
+		ctx = obsrecv.StartMetricsOp(ctx)
+		l, err := logsUnmarshaler.UnmarshalLogs(token)
+		if err != nil {
+			obsrecv.EndLogsOp(ctx, typeStr, 0, err)
+		} else {
+			err = logs.ConsumeLogs(ctx, l)
+			obsrecv.EndLogsOp(ctx, typeStr, l.LogRecordCount(), err)
+		}
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &receiver{input: input, id: configuration.ID()}, nil
+}
+
+func createMetricsReceiver(_ context.Context, settings component.ReceiverCreateSettings, configuration config.Receiver, metrics consumer.Metrics) (component.MetricsReceiver, error) {
+	metricsUnmarshaler := pmetric.NewJSONUnmarshaler()
+	obsrecv := obsreport.NewReceiver(obsreport.ReceiverSettings{
+		ReceiverID:             configuration.ID(),
+		Transport:              transport,
+		ReceiverCreateSettings: settings,
+	})
+	input, err := configuration.(*cfg).Config.Build(settings.Logger.Sugar(), func(ctx context.Context, attrs *fileconsumer.FileAttributes, token []byte) {
+		ctx = obsrecv.StartMetricsOp(ctx)
+		m, err := metricsUnmarshaler.UnmarshalMetrics(token)
+		if err != nil {
```

Comment:
it looks really hard to reuse the same Input object across logs, metrics and traces since they have a complete config object.
I like the idea of being a bit more intelligent so we don't log a bunch of errors when we have a mixed case. I wonder if we can make it a simple substring check to reduce the load.

Reply:
I think it would be reasonable to capture this as an issue for now and accept the component with the noise for now. This should probably be addressed in some form before moving the component to beta though.

Reply:
I am not satisfied with the current state, I still need to fix the config first. I'll keep thinking.

Reply:
We gotta land this PR, so I guess we can live with an issue for now. I updated the config model.